packages/material-ui/src/Typography/Typography.js

```diff
@@ -215,43 +235,32 @@ Typography.propTypes = {
   /**
    * Applies the theme typography styles.
    */
-  variant: PropTypes.oneOf([
-    'body1',
-    'body2',
-    'button',
-    'caption',
-    'h1',
-    'h2',
-    'h3',
-    'h4',
-    'h5',
-    'h6',
-    'inherit',
-    'overline',
-    'subtitle1',
-    'subtitle2',
+  variant: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
+    PropTypes.oneOf([
+      'body1',
+      'body2',
+      'button',
+      'caption',
+      'h1',
+      'h2',
+      'h3',
+      'h4',
+      'h5',
+      'h6',
+      'inherit',
+      'overline',
+      'subtitle1',
+      'subtitle2',
+    ]),
+    PropTypes.string,
   ]),
   /**
    * The component maps the variant prop to a range of different HTML element types.
    * For instance, subtitle1 to `<h6>`.
    * If you wish to change that mapping, you can provide your own.
    * Alternatively, you can use the `component` prop.
    */
-  variantMapping: PropTypes.shape({
-    body1: PropTypes.string,
-    body2: PropTypes.string,
-    button: PropTypes.string,
-    caption: PropTypes.string,
-    h1: PropTypes.string,
-    h2: PropTypes.string,
-    h3: PropTypes.string,
-    h4: PropTypes.string,
-    h5: PropTypes.string,
-    h6: PropTypes.string,
-    overline: PropTypes.string,
-    subtitle1: PropTypes.string,
-    subtitle2: PropTypes.string,
-  }),
+  variantMapping: PropTypes /* @typescript-to-proptypes-ignore */.object,
```

Comment:
With adding new variants new keys can be added here, so this was relaxed to `object`.